go/arrow/flight/record_batch_reader.go

```diff
@@ -234,3 +234,30 @@ func StreamChunksFromReader(rdr array.RecordReader, ch chan<- StreamChunk) {
 		}
 	}
 }
+
+func ConcatenateReaders(rdrs []array.RecordReader, ch chan<- StreamChunk) {
+	defer close(ch)
+	defer func() {
+		for _, r := range rdrs {
+			r.Release()
+		}
+
+		if err := recover(); err != nil {
+			ch <- StreamChunk{Err: fmt.Errorf("panic while reading: %s", err)}
```

Comment:
yikes, is a panic typical here?

Reply:
it shouldn't be, but it is possible so i'm protecting against other server writers throwing panics in their implementation by catching it and returning an error instead of bringing down the server.